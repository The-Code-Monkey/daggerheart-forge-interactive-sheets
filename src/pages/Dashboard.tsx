import { JSX, useEffect, useState } from "react";
import {
  Card,
  CardContent,
  CardDescription,
  CardHeader,
  CardTitle,
} from "@/components/ui/card";
import { Button } from "@/components/ui/button";
import { Badge } from "@/components/ui/badge";
import {
  AlertDialog,
  AlertDialogAction,
  AlertDialogCancel,
  AlertDialogContent,
  AlertDialogDescription,
  AlertDialogFooter,
  AlertDialogHeader,
  AlertDialogTitle,
  AlertDialogTrigger,
} from "@/components/ui/alert-dialog";
import {
  Users,
  Sword,
  BookOpen,
  Plus,
  LogOut,
  Eye,
  Pencil,
  Beer,
  Trash2,
} from "lucide-react";
import { Link } from "react-router-dom";
import { useAuth } from "@/contexts/AuthContext";
import { useToast } from "@/hooks/use-toast";
import { getCharacters } from "@/integrations/supabase/helpers";
import { CharacterWithRelations } from "@/lib/types";
import { supabase } from "@/integrations/supabase/client";

const Dashboard = (): JSX.Element => {
  const { user, signOut } = useAuth();
  const { toast } = useToast();

  const [characters, setCharacters] = useState<CharacterWithRelations[]>([]);

  useEffect(() => {
    const fetchCharacters = async () => {
      const data = await getCharacters(String(user?.id));

      if (data) {
        setCharacters(data);
      }
    };

    void fetchCharacters();
  }, [user]);

  const handleSignOut = async () => {
    await signOut();
    toast({
      title: "Signed out",
      description: "You have been successfully signed out.",
    });
  };

  const handleDeleteCharacter = async (
    characterId: string,
    characterName: string,
  ) => {
    try {
      const { error } = await supabase
        .from("characters")
        .delete()
        .eq("id", characterId);

      if (error) {
        toast({
          title: "Error",
          description: "Failed to delete character. Please try again.",
          variant: "destructive",
        });
        return;
      }

      // Remove the character from the local state
      setCharacters((prev) => prev.filter((char) => char.id !== characterId));

      toast({
        title: "Character deleted",
        description: `${characterName} has been successfully deleted.`,
      });
    } catch (error) {
      console.error("Error deleting character:", error);
      toast({
        title: "Error",
        description: "An unexpected error occurred. Please try again.",
        variant: "destructive",
      });
    }
  };

  const characterCount = characters.length;
  const maxCharacters = 5;
  const canCreateMore = characterCount < maxCharacters;

  return (
    <div className="min-h-screen bg-nebula p-4">
      <div className="max-w-6xl mx-auto">
        {/* Header */}
        <div className="flex justify-between items-center mb-8">
          <div>
            <h1 className="text-4xl font-bold text-white mb-2">
              Welcome back, {user?.user_metadata.username ?? "Adventurer"}!
            </h1>
            <p className="text-brand-200">
              Manage your characters and campaigns
            </p>
          </div>
          <div className="flex flex-row gap-2">
            <Button asChild>
              <Link to="/homebrew">
                <Beer className="w-4 h-4 mr-2" />
                Homebrew
              </Link>
            </Button>
            <Button
              onClick={() => {
                void handleSignOut();
              }}
              variant="outline"
              className="border-brand-400 text-brand-100"
            >
              <LogOut className="w-4 h-4 mr-2" />
              Sign Out
            </Button>
          </div>
        </div>

        {/* Character Overview */}
        <div className="grid grid-cols-1 md:grid-cols-3 gap-6 mb-8">
          <Card>
            <CardHeader className="flex flex-row items-center justify-between space-y-0 pb-2">
              <CardTitle className="text-sm font-medium text-white">
                Characters
              </CardTitle>
              <Users className="h-4 w-4 text-yellow-400" />
            </CardHeader>
            <CardContent>
              <div className="text-2xl font-bold text-white">
                {characterCount}/{maxCharacters}
              </div>
              <p className="text-xs text-brand-200">
                {canCreateMore
                  ? `${String(maxCharacters - characterCount)} slots remaining`
                  : "Maximum reached"}
              </p>
            </CardContent>
          </Card>

          <Card>
            <CardHeader className="flex flex-row items-center justify-between space-y-0 pb-2">
              <CardTitle className="text-sm font-medium text-white">
                Campaigns
              </CardTitle>
              <Sword className="h-4 w-4 text-yellow-400" />
            </CardHeader>
            <CardContent>
              <div className="text-2xl font-bold text-white">Coming Soon</div>
              <p className="text-xs text-brand-200">Campaign management</p>
            </CardContent>
          </Card>

          <Card>
            <CardHeader className="flex flex-row items-center justify-between space-y-0 pb-2">
              <CardTitle className="text-sm font-medium text-white">
                Rule Books
              </CardTitle>
              <BookOpen className="h-4 w-4 text-yellow-400" />
            </CardHeader>
            <CardContent>
              <div className="text-2xl font-bold text-white">Coming Soon</div>
              <p className="text-xs text-brand-200">Digital rule access</p>
            </CardContent>
          </Card>
        </div>

        {/* Characters Section */}
        <Card>
          <CardHeader className="flex flex-row items-center justify-between">
            <div>
              <CardTitle className="text-white text-xl">
                Your Characters
              </CardTitle>
              <CardDescription className="text-brand-200">
                Manage your Daggerheart characters
              </CardDescription>
            </div>
            {canCreateMore && (
              <Link to="/character-builder">
                <Button className="bg-yellow-500 hover:bg-yellow-600 text-black font-semibold">
                  <Plus className="w-4 h-4 mr-2" />
                  Create Character
                </Button>
              </Link>
            )}
          </CardHeader>
          <CardContent>
            {characters.length > 0 ? (
              <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-4">
                {characters.map((character) => {
                  const link = character.complete
                    ? `/character-sheet/${character.id}`
                    : `/character-builder/${character.id}`;

                  return (
                    <Card
                      key={character.id}
                      className="bg-slate-800/30 border-brand-500/20"
                    >
                      <CardHeader className="pb-2">
                        <CardTitle className="text-white text-lg">
                          {character.name}
                        </CardTitle>
                        <div className="flex gap-2">
                          <Badge variant="secondary" className="text-xs">
                            Level {character.level}
                          </Badge>
                          {character.ancestry && (
                            <Badge
                              variant="outline"
                              className="text-xs border-brand-400 text-brand-200"
                            >
                              {character.ancestry.name ?? "Unknown"}
                            </Badge>
                          )}
                        </div>
                      </CardHeader>
                      <CardContent>
                        <p className="text-brand-200 text-sm mb-3">
                          {character.class?.name ?? "No class selected"}
                          {character.subclass
                            ? ` - (${String(character.subclass.name)})`
                            : ""}
                        </p>
                        <div className="flex gap-2">
                          <Link to={link} className="flex-1">
                            <Button
                              size="sm"
                              className="w-full bg-blue-600 hover:bg-blue-700 text-white"
                            >
                              {character.complete ? (
                                <>
                                  <Eye className="w-4 h-4 mr-2" />
                                  View Sheet
                                </>
                              ) : (
                                <>
                                  <Pencil className="w-4 h-4 mr-2" />
                                  Edit Sheet
                                </>
                              )}
                            </Button>
                          </Link>
                          <AlertDialog>
                            <AlertDialogTrigger asChild>
                              <Button
                                size="sm"
                                variant="destructive"
                                className="px-3"
                              >
                                <Trash2 className="w-4 h-4" />
                              </Button>
                            </AlertDialogTrigger>
                            <AlertDialogContent>
                              <AlertDialogHeader>
                                <AlertDialogTitle>
                                  Delete Character
                                </AlertDialogTitle>
                                <AlertDialogDescription>
                                  Are you sure you want to delete "
                                  {character.name}"? This action cannot be
                                  undone.
                                </AlertDialogDescription>
                              </AlertDialogHeader>
                              <AlertDialogFooter>
                                <AlertDialogCancel>Cancel</AlertDialogCancel>
                                <AlertDialogAction
                                  onClick={() =>
                                    void handleDeleteCharacter(
                                      character.id,
                                      character.name,
                                    )
                                  }
                                  className="bg-red-600 hover:bg-red-700"
                                >
                                  Delete
                                </AlertDialogAction>
                              </AlertDialogFooter>
                            </AlertDialogContent>
                          </AlertDialog>
                        </div>
                      </CardContent>
                    </Card>
                  );
                })}
              </div>
            ) : (
              <div className="text-center py-8">
                <p className="text-brand-200 mb-4">
                  You haven't created any characters yet.
                </p>
                {canCreateMore && (
                  <Link to="/character-builder">
                    <Button className="bg-yellow-500 hover:bg-yellow-600 text-black font-semibold">
                      <Plus className="w-4 h-4 mr-2" />
                      Create Your First Character
                    </Button>
                  </Link>
                )}
              </div>
            )}

            {!canCreateMore && (
              <div className="mt-4 p-4 bg-orange-900/30 border border-orange-500/30 rounded-lg">
                <p className="text-orange-200 text-sm">
                  You've reached the maximum of {maxCharacters} characters.
                  Delete a character to create a new one.
                </p>
              </div>
            )}
          </CardContent>
        </Card>

        {/* Coming Soon Features */}
        <div className="grid grid-cols-1 md:grid-cols-2 gap-6 mt-8">
          <Card className="bg-linear-to-br from-slate-800/40 to-gray-800/40 border-gray-500/30 opacity-60">
            <CardHeader>
              <CardTitle className="text-white">Campaign Management</CardTitle>
              <CardDescription className="text-gray-300">
                Create and manage your Daggerheart campaigns
              </CardDescription>
            </CardHeader>
            <CardContent>
              <Badge variant="secondary" className="bg-gray-600">
                Coming Soon
              </Badge>
            </CardContent>
          </Card>

          <Card className="bg-linear-to-br from-slate-800/40 to-gray-800/40 border-gray-500/30 opacity-60">
            <CardHeader>
              <CardTitle className="text-white">Digital Dice & Tools</CardTitle>
              <CardDescription className="text-gray-300">
                Dice rolling, initiative tracking, and more
              </CardDescription>
            </CardHeader>
            <CardContent>
              <Badge variant="secondary" className="bg-gray-600">
                Coming Soon
              </Badge>
            </CardContent>
          </Card>
        </div>
      </div>
    </div>
  );
};

export default Dashboard;
