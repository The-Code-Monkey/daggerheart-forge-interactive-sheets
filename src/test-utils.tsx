import React from "react";
import { render, RenderOptions } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

// Mock AuthContext
const mockAuthContext = {
  user: null,
  loading: false,
  signIn: jest.fn(),
  signOut: jest.fn(),
  signUp: jest.fn(),
};

const AuthProvider = ({ children }: { children: React.ReactNode }) => {
  return <div data-testid="auth-provider">{children}</div>;
};

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <BrowserRouter>
      <AuthProvider>{children}</AuthProvider>
    </BrowserRouter>
  );
};

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

// Export everything from testing-library/react
export * from "@testing-library/react";
export { customRender as render };

// Mock useAuth hook
export const mockUseAuth = () => mockAuthContext;
