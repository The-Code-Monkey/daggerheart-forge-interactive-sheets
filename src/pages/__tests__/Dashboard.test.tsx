import { render, screen } from '../../test-utils';
import Dashboard from '../Dashboard';

// Mock useAuth hook
jest.mock('@/contexts/AuthContext', () => ({
  useAuth: () => ({
    user: { id: '1', email: 'test@example.com' },
    loading: false
  })
}));

describe('Dashboard', () => {
  it('renders dashboard content', () => {
    render(<Dashboard />);
    // Add specific assertions based on Dashboard content
    expect(screen.getByText(/dashboard/i)).toBeInTheDocument();
  });
});
