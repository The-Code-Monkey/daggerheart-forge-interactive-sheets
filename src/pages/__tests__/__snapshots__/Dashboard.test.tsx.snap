// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Dashboard renders correctly 1`] = `
<div
  data-testid="auth-provider"
>
  <div
    class="min-h-screen bg-nebula p-4"
  >
    <div
      class="max-w-6xl mx-auto"
    >
      <div
        class="flex justify-between items-center mb-8"
      >
        <div>
          <h1
            class="text-4xl font-bold text-white mb-2"
          >
            Welcome back, 
            Adventurer
            !
          </h1>
          <p
            class="text-brand-200"
          >
            Manage your characters and campaigns
          </p>
        </div>
        <div
          class="flex flex-row gap-2"
        >
          <a
            href="/homebrew"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-beer w-4 h-4 mr-2"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17 11h1a3 3 0 0 1 0 6h-1"
              />
              <path
                d="M9 12v6"
              />
              <path
                d="M13 12v6"
              />
              <path
                d="M14 7.5c-1 0-1.44.5-3 .5s-2-.5-3-.5-1.72.5-2.5.5a2.5 2.5 0 0 1 0-5c.78 0 1.57.5 2.5.5S9.44 2 11 2s2 1.5 3 1.5 1.72-.5 2.5-.5a2.5 2.5 0 0 1 0 5c-.78 0-1.5-.5-2.5-.5Z"
              />
              <path
                d="M5 8v12a2 2 0 0 0 2 2h8a2 2 0 0 0 2-2V8"
              />
            </svg>
            Homebrew
          </a>
          <button
            class="inline-flex items-center hover:cursor-pointer justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors duration-300 ease-in-out focus-visible:outline-hidden focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border bg-brand-700/80 hover:bg-brand-700 hover:text-accent-foreground h-10 px-4 py-2 border-brand-400 text-brand-100"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-log-out w-4 h-4 mr-2"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m16 17 5-5-5-5"
              />
              <path
                d="M21 12H9"
              />
              <path
                d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
              />
            </svg>
            Sign Out
          </button>
        </div>
      </div>
      <div
        class="grid grid-cols-1 md:grid-cols-3 gap-6 mb-8"
      >
        <div
          class="rounded-lg border text-card-foreground shadow-sm bg-gradient-to-br from-slate-800/80 to-slate-900/80 border-brand-500/30 backdrop-blur-sm"
        >
          <div
            class="p-6 flex flex-row items-center justify-between space-y-0 pb-2"
          >
            <h3
              class="tracking-tight text-sm font-medium text-white"
            >
              Characters
            </h3>
            <svg
              aria-hidden="true"
              class="lucide lucide-users h-4 w-4 text-yellow-400"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"
              />
              <path
                d="M16 3.128a4 4 0 0 1 0 7.744"
              />
              <path
                d="M22 21v-2a4 4 0 0 0-3-3.87"
              />
              <circle
                cx="9"
                cy="7"
                r="4"
              />
            </svg>
          </div>
          <div
            class="p-6 pt-0"
          >
            <div
              class="text-2xl font-bold text-white"
            >
              0
              /
              5
            </div>
            <p
              class="text-xs text-brand-200"
            >
              5 slots remaining
            </p>
          </div>
        </div>
        <div
          class="rounded-lg border text-card-foreground shadow-sm bg-gradient-to-br from-slate-800/80 to-slate-900/80 border-brand-500/30 backdrop-blur-sm"
        >
          <div
            class="p-6 flex flex-row items-center justify-between space-y-0 pb-2"
          >
            <h3
              class="tracking-tight text-sm font-medium text-white"
            >
              Campaigns
            </h3>
            <svg
              aria-hidden="true"
              class="lucide lucide-sword h-4 w-4 text-yellow-400"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="14.5 17.5 3 6 3 3 6 3 17.5 14.5"
              />
              <line
                x1="13"
                x2="19"
                y1="19"
                y2="13"
              />
              <line
                x1="16"
                x2="20"
                y1="16"
                y2="20"
              />
              <line
                x1="19"
                x2="21"
                y1="21"
                y2="19"
              />
            </svg>
          </div>
          <div
            class="p-6 pt-0"
          >
            <div
              class="text-2xl font-bold text-white"
            >
              Coming Soon
            </div>
            <p
              class="text-xs text-brand-200"
            >
              Campaign management
            </p>
          </div>
        </div>
        <div
          class="rounded-lg border text-card-foreground shadow-sm bg-gradient-to-br from-slate-800/80 to-slate-900/80 border-brand-500/30 backdrop-blur-sm"
        >
          <div
            class="p-6 flex flex-row items-center justify-between space-y-0 pb-2"
          >
            <h3
              class="tracking-tight text-sm font-medium text-white"
            >
              Rule Books
            </h3>
            <svg
              aria-hidden="true"
              class="lucide lucide-book-open h-4 w-4 text-yellow-400"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 7v14"
              />
              <path
                d="M3 18a1 1 0 0 1-1-1V4a1 1 0 0 1 1-1h5a4 4 0 0 1 4 4 4 4 0 0 1 4-4h5a1 1 0 0 1 1 1v13a1 1 0 0 1-1 1h-6a3 3 0 0 0-3 3 3 3 0 0 0-3-3z"
              />
            </svg>
          </div>
          <div
            class="p-6 pt-0"
          >
            <div
              class="text-2xl font-bold text-white"
            >
              Coming Soon
            </div>
            <p
              class="text-xs text-brand-200"
            >
              Digital rule access
            </p>
          </div>
        </div>
      </div>
      <div
        class="rounded-lg border text-card-foreground shadow-sm bg-gradient-to-br from-slate-800/80 to-slate-900/80 border-brand-500/30 backdrop-blur-sm"
      >
        <div
          class="space-y-1.5 p-6 flex flex-row items-center justify-between"
        >
          <div>
            <h3
              class="font-semibold tracking-tight text-white text-xl"
            >
              Your Characters
            </h3>
            <p
              class="text-sm text-brand-200"
            >
              Manage your Daggerheart characters
            </p>
          </div>
          <a
            href="/character-builder"
          >
            <button
              class="inline-flex items-center hover:cursor-pointer justify-center gap-2 whitespace-nowrap rounded-md text-sm ring-offset-background transition-colors duration-300 ease-in-out focus-visible:outline-hidden focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 h-10 px-4 py-2 bg-yellow-500 hover:bg-yellow-600 text-black font-semibold"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-plus w-4 h-4 mr-2"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 12h14"
                />
                <path
                  d="M12 5v14"
                />
              </svg>
              Create Character
            </button>
          </a>
        </div>
        <div
          class="p-6 pt-0"
        >
          <div
            class="text-center py-8"
          >
            <p
              class="text-brand-200 mb-4"
            >
              You haven't created any characters yet.
            </p>
            <a
              href="/character-builder"
            >
              <button
                class="inline-flex items-center hover:cursor-pointer justify-center gap-2 whitespace-nowrap rounded-md text-sm ring-offset-background transition-colors duration-300 ease-in-out focus-visible:outline-hidden focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 h-10 px-4 py-2 bg-yellow-500 hover:bg-yellow-600 text-black font-semibold"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-plus w-4 h-4 mr-2"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 12h14"
                  />
                  <path
                    d="M12 5v14"
                  />
                </svg>
                Create Your First Character
              </button>
            </a>
          </div>
        </div>
      </div>
      <div
        class="grid grid-cols-1 md:grid-cols-2 gap-6 mt-8"
      >
        <div
          class="rounded-lg border text-card-foreground shadow-sm bg-gradient-to-br backdrop-blur-sm bg-linear-to-br from-slate-800/40 to-gray-800/40 border-gray-500/30 opacity-60"
        >
          <div
            class="flex flex-col space-y-1.5 p-6"
          >
            <h3
              class="text-2xl font-semibold leading-none tracking-tight text-white"
            >
              Campaign Management
            </h3>
            <p
              class="text-sm text-gray-300"
            >
              Create and manage your Daggerheart campaigns
            </p>
          </div>
          <div
            class="p-6 pt-0"
          >
            <div
              class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-hidden focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent text-secondary-foreground hover:bg-secondary/80 bg-gray-600"
            >
              Coming Soon
            </div>
          </div>
        </div>
        <div
          class="rounded-lg border text-card-foreground shadow-sm bg-gradient-to-br backdrop-blur-sm bg-linear-to-br from-slate-800/40 to-gray-800/40 border-gray-500/30 opacity-60"
        >
          <div
            class="flex flex-col space-y-1.5 p-6"
          >
            <h3
              class="text-2xl font-semibold leading-none tracking-tight text-white"
            >
              Digital Dice & Tools
            </h3>
            <p
              class="text-sm text-gray-300"
            >
              Dice rolling, initiative tracking, and more
            </p>
          </div>
          <div
            class="p-6 pt-0"
          >
            <div
              class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-hidden focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent text-secondary-foreground hover:bg-secondary/80 bg-gray-600"
            >
              Coming Soon
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
